src/Symfony/Component/TypeInfo/TypeResolver/StringTypeResolver.php

```diff
@@ -58,10 +59,21 @@ final class StringTypeResolver implements TypeResolverInterface
      */
     private static array $classExistCache = [];
 
+    private readonly Lexer $lexer;
+    private readonly TypeParser $parser;
+
     public function __construct(
-        private Lexer $lexer = new Lexer(),
-        private TypeParser $parser = new TypeParser(new ConstExprParser()),
+        ?Lexer $lexer = null,
+        ?TypeParser $typeParser = null,
```

Comment:
Those parameters are basically ignored now. Is that what we want?

Reply:
Indeed, my bad